keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -313,20 +337,57 @@ has completed parsing its `kubeconfig`, flags and environment variables).
 This allows the executable to perform different actions based on the current
 cluster (i.e. get a token for a particular cluster).  The `Cluster` struct is
 flexible in that it not only provides all details required to communicate with
-the cluster (hostname and TLS config), but that is also allows arbitrary
+the cluster as one would via a `kubeconfig` (i.e. everything from
+`"k8s.io/client-go/tools/clientcmd/api/v1".Cluster`), but also allows arbitrary
 per-cluster configuration to be passed to the executable via the `config` field.
 This field can contain arbitrary data that is passed to the executable without
 modification.  This allows extra user-defined data (i.e. an OAuth client ID for
 audience scoping) to be passed through the `spec.cluster` field.  The user
-configures this via the `kubeconfig`'s `clusters.[...].cluster.extensions[exec].extension`
-field.  The `exec` named extension is reserved for this purpose.
+configures this via the `kubeconfig`'s `clusters.[...].cluster.extensions[client.authentication.k8s.io/exec].extension`
+field.  The `client.authentication.k8s.io/exec` named extension is reserved for this
+purpose.
+
+To ensure that the `Cluster` struct maintains the same cluster connection
+capabilities as one gets from a `kubeconfig`, the `Cluster` fields (Go and JSON) must
+be named the same as `"k8s.io/client-go/tools/clientcmd/api/v1".Cluster`, with the
+exception of `InsecureSkipTLSVerify`, which has been left out to discourage its use, and
```

Comment:
Thinking about this more, I think we should include `InsecureSkipTLSVerify`.

If the kubeconfig has `insecure-skip-tls-verify: true` set, whatever credentials the auth plugin returns will be happily sent by client-go without verifying TLS.

If we omit this, I anticipate some auth plugins trying connections with insecure TLS as a fallback "just in case" that's how the user's kubeconfig was set up.

One upside of including `InsecureSkipTLSVerify` info is that it would allow credential providers to detect if replayable tokens they planned to return would be used in an insecure context, and to refuse.

Reply:
ACK. @enj may slay me when he returns, but I'll add this into the struct. :)

Reply:
> One upside of including `InsecureSkipTLSVerify` info is that it would allow credential providers to detect if replayable tokens they planned to return would be used in an insecure context, and to refuse.

That's how I talked myself into this

Reply:
You speak truth!